tests/infrastructure/certificates.go

```diff
@@ -75,17 +74,12 @@ var _ = DescribeInfra("[rfe_id:4102][crit:medium][vendor:cnv-qe@redhat.com][leve
 		}, 10*time.Second, 1*time.Second).Should(BeNumerically(">", 0))
 
 		By("destroying the certificate")
-		err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
-			secret, err := virtClient.CoreV1().Secrets(flags.KubeVirtInstallNamespace).Get(context.Background(), components.KubeVirtCASecretName, metav1.GetOptions{})
-			Expect(err).ToNot(HaveOccurred())
-			secret.Data = map[string][]byte{
-				"tls.crt": []byte(""),
-				"tls.key": []byte(""),
-			}
-
-			_, err = virtClient.CoreV1().Secrets(flags.KubeVirtInstallNamespace).Update(context.Background(), secret, metav1.UpdateOptions{})
-			return err
-		})
+		secretPatch, err := patch.New(
+			patch.WithReplace("/data/tls.crt", ""),
+			patch.WithReplace("/data/tls.key", ""),
+		).GeneratePayload()
+		Expect(err).ToNot(HaveOccurred())
+		_, err = virtClient.CoreV1().Secrets(flags.KubeVirtInstallNamespace).Patch(context.Background(), components.KubeVirtCASecretName, types.JSONPatchType, secretPatch, metav1.PatchOptions{})
```

Comment:
It can still fail due to "object was changed", the question is if the retry was just because it was update
or because here it happens in the middle of other orthogonal updates
and then worth please to keep the retry imho, wdyt?

Reply:
IMO - if the object was changed, then we should understand why, and "retryOnConflict" is only masking it. Better fail than retrying.

Reply:
some of the conflicts are totally legit if the object is under changing, for example during its creation, there might multi entities that can update it
we can postpone and be proactive, but it can introduce flakes, worth maybe to check why the Retry was added, and when, the history might teach on the reason

Reply:
I see that specifically in `tests/infrastructure/certificates.go` the secret `components.KubeVirtCASecretName` is modified only here. Either use here a new secret to test it, or check if this secret was updated in other tests to address @oshoval's concerns.

Reply:
IIRC - Unless there is a bug there should not be a retry.
* the crt and key are either valid (meaning the certs were successfuly rotated) or are empty/outdated/corrupt - then they should be rotated. This test checks that the rotation happened by forcing it to rotate.
* cert-manager should not change the key/crt if they are valid, so (unless there is a bug, which this test should catch)) the only chance to conflict is if the certs rotate

Reply:
@codingben I don't follow you. 
In this test, we:
1. delete the crt+key in the secret
2. wait until cert-manager rotates then, updating crt+key
3. externally calculate and make sure the cert-manager did its job well..

To address your comment:
>Either use here a new secret to test it

This secret is created by virt-operator and controlled by cert-manager. We specifically want to make sure that secret is rotated, not a new one

> or check if this secret was updated in other tests to address @oshoval's concerns.

why do we need to check if this secret was updated? Do you mean parallel tests? please elaborate

Reply:
Got it, thanks.

> Is your concern conflicts that come from parallel tests?

Yes, exactly.

Reply:
@codingben then perhaps we should serial this test then. WDYT?

Reply:
@RamLavi I'm not sure, because `Patch` could be not have the same behavior as `Update`? @oshoval WDYT?

Reply:
I think @oshoval specifically raised concerns related to `Update` and `Patch` may not update ID of the resource and not cause conflicts.

Reply:
I think that i was wrong, and patch is atomic, as was discussed offline

Reply:
For the question of "can patch conflict, and if it can, is it only for the same field" - I didn't find yet a full answer.
Did you learn otherwise?

Reply:
I even think that this wont fail, because there is no resourceID in the request, hence it can't know what to compare to, it will just replace or merge, according the `patchStrategy` (default is replace at least on Strategic merge which is the default iiuc)
If one want to do a safe patch, (which set only if the current value match a specific input)
there is `patch.PatchTestOp`

One can simulate it easily if desired by Getting the resource, and then do 2 patches on the same field

Reply:
I don't think it matters if the ID is sent or not.
the patch  operation doesn't hold it but inside they have to 
1.  do a GET request
2.  merge accordingly to the patch operation (add/replace/delete..)
3. do a PUT request (which goes to etcd)

if by the time it gets to etcd the ID is already changed - then there will be a conflict. Is my logic wrong?

>One can simulate it easily if desired by getting the resource, and then do 2 patches on the same field

I actually did, trying to patch different paths, but didn't get conflicts..  But's not a good enough proof for me.

Reply:
> I don't think it matters if the ID is sent or not. the patch operation doesn't hold it but inside they have to
> 
> 1. do a GET request
> 2. merge accordingly to the patch operation (add/replace/delete..)
> 3. do a PUT request (which goes to etcd)
> 
> if by the time it gets to etcd the ID is already changed - then there will be a conflict. Is my logic wrong?

I don't know to say this way or another, maybe best we ask on k8s users slack

> 
> > One can simulate it easily if desired by getting the resource, and then do 2 patches on the same field
> 
> I actually did, trying to patch different paths, but didn't get conflicts.. But's not a good enough proof for me.

Reply:
anyhow imho we shouldnt block any of those PRs about that, because even if there is a conflict, it is fine to just start without the retry and add it according needs later
best to allow research to be done in parallel whenever possible
thanks